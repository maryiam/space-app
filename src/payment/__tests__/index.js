import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { SelectPayment } from '../';

describe('payment/', () => {
  describe('#render', () => {
    it('renders correctly with no information', () => {
      const tree = renderer.create(<SelectPayment actions={{}} />);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when loading', () => {
      const tree = renderer.create(<SelectPayment loading actions={{}} />);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with currencies available', () => {
      const tree = renderer.create(
        <SelectPayment
          currencies={[
            { id: 0, name: 'currency 1' },
            { id: 1, name: 'currency 2' }
          ]}
          actions={{}}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with no currency available', () => {
      const tree = renderer.create(
        <SelectPayment
          currencies={[]}
          actions={{}}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
