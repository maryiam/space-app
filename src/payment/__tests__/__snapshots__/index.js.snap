// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payment/ #render renders correctly when loading 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: 'Passion One',sans-serif;
  font-weight: normal;
  margin-top: 0;
  margin-bottom: 1rem;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #8a8894;
  margin-bottom: 1rem;
}

<section
  className="c0"
>
  <h2
    className="c1"
  >
    Payment Mean
  </h2>
  <span
    className="c2"
  >
    Loading...
  </span>
</section>
`;

exports[`payment/ #render renders correctly with currencies available 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: 'Passion One',sans-serif;
  font-weight: normal;
  margin-top: 0;
  margin-bottom: 1rem;
}

.c2 {
  display: block;
}

.c2 + .c2 {
  margin-top: 1rem;
}

.c3 {
  display: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4::before {
  content: '';
  display: block;
  width: 1em;
  height: 1em;
  background-color: #6f7984;
  box-shadow: inset -3px -5px 15px rgba(0,0,0,0.75);
  border-radius: 50%;
  margin-right: 1rem;
}

input:checked + .c4::before {
  background-image: radial-gradient( circle, rgba(0,0,0,0.65) 50%, #6f7984 50% );
  background-repeat: no-repeat;
  background-size: 10px;
  background-position: 1px 0px;
}

<section
  className="c0"
>
  <h2
    className="c1"
  >
    Payment Mean
  </h2>
  <label
    className="c2"
    htmlFor="currency-0"
  >
    <input
      className="c3"
      id="currency-0"
      name="currency"
      onChange={[Function]}
      type="radio"
      value={0}
    />
    <span
      className="c4"
    >
      currency 1
    </span>
  </label>
  <label
    className="c2"
    htmlFor="currency-1"
  >
    <input
      className="c3"
      id="currency-1"
      name="currency"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    <span
      className="c4"
    >
      currency 2
    </span>
  </label>
</section>
`;

exports[`payment/ #render renders correctly with no currency available 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: 'Passion One',sans-serif;
  font-weight: normal;
  margin-top: 0;
  margin-bottom: 1rem;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #8a8894;
  margin-bottom: 1rem;
}

<section
  className="c0"
>
  <h2
    className="c1"
  >
    Payment Mean
  </h2>
  <span
    className="c2"
  >
    No currency available for this destination
  </span>
</section>
`;

exports[`payment/ #render renders correctly with no information 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-family: 'Passion One',sans-serif;
  font-weight: normal;
  margin-top: 0;
  margin-bottom: 1rem;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #8a8894;
  margin-bottom: 1rem;
}

<section
  className="c0"
>
  <h2
    className="c1"
  >
    Payment Mean
  </h2>
  <span
    className="c2"
  >
    Select a destination to view available currencies
  </span>
</section>
`;
